internal/convert/converter_test.go

```diff
@@ -896,13 +902,100 @@ func TestConverter(t *testing.T) { //nolint:maintidx
 			to:          pointer(OuterStruct{}),
 			err:         "'' expected a map, got 'string'",
 		},
-		// unsupported.
 		{
-			description: "to interface (unsupported)",
+			description: "int to interface",
+			from:        42,
+			to:          pointer(any(nil)),
+			expected:    pointer(any(42)),
+		},
+		{
+			description: "string to interface",
 			from:        "str",
 			to:          pointer(any(nil)),
-			err:         ": unsupported type: interface",
+			expected:    pointer(any("str")),
+		},
+		{
+			description: "float to interface",
+			from:        42.42,
+			to:          pointer(any(nil)),
+			expected:    pointer(any(42.42)),
+		},
+		{
+			description: "map to interface",
+			from:        map[string]int{"key": 42, "keySensitive": 43},
+			to:          pointer(any(nil)),
+			expected:    pointer(any(map[string]int{"key": 42, "keySensitive": 43})),
+		},
+		{
+			description: "map to interface (with keyMap)", // Probably redundant.
+			opts: []convert.Option{
+				convert.WithKeyMapper(strings.ToLower),
+			},
+			from:     map[string]int{"key": 42, "keysensitive": 43},
+			to:       pointer(any(nil)),
+			expected: pointer(any(map[string]int{"key": 42, "keysensitive": 43})),
+		},
+		{
+			description: "packed KV and field to map[string]interface{}",
+			from: map[string]interface{}{
+				"key1": maps.KeyValue{
+					Key:   "key1",
+					Value: "value1",
+				},
+				"key2": "value2",
+			},
+			to: pointer(map[string]interface{}{}),
+			expected: pointer(map[string]interface{}{
+				"key1": "value1",
+				"key2": "value2",
+			}),
+		},
+		{
+			description: "packed KV and field to struct (with keyMap)",
+			opts: []convert.Option{
+				convert.WithKeyMapper(strings.ToLower),
+			},
+			from: map[string]interface{}{
+				"key1": maps.KeyValue{
+					Key:   "key1",
+					Value: "value1",
+				},
+				"key2": "value2",
+			},
+			to: pointer(struct {
+				Key1 interface{}
+				Key2 interface{}
+			}{}),
+			expected: pointer(struct {
+				Key1 interface{}
+				Key2 interface{}
+			}{
+				Key1: "value1",
+				Key2: "value2",
+			}),
+		},
+		{
+			description: "packed KV and field to interface{}",
```

Comment:
This test case shows that special handling is necessary. Now, all complex data types will be treated as map[string]interface{}.